### ISSUE TITLE
`duration_suboptimal_units` should not trigger on small values

### ISSUE THREAD

**Issue by ensc** [rust-lang/rust-clippy#16532]
### Summary

The `duration_suboptimal_units` lint triggers on 

```
Duration::from_millis(1_000);
``` 

suggesting

```
Duration::from_secs(1)
````

but this makes quick adjustments harder. E.g. Changing 1_000 to 1_200 requires modifying both the function name and the value.

I suggest that the lint triggers only when the target value exceeds a certain threshold (e.g. 10).  

### Rationale

- code becomes inconsistent; having

  ```
  let _ = Duration::from_secs(210);
  let _ = Duration::from_mins(3);
  let _ = Duration::from_secs(150);

  let _ = Duration::from_millis(800);
  let _ = Duration::from_secs(1);
  let _ = Duration::from_millis(1_100);
  ```

  breaks readability. Readers must mentally convert between units.

- minimal adaptions of the duration require (non-trivial) code changes

- `from_secs(1)` is not easier to read than `from_millis(1_000)`
 
- code smell like `from_millis(1_001)` to workaround the lint

### Lint Name

`duration_suboptimal_units`

### Version

```text
clippy 0.1.95 (a293cc4af8 2026-01-30)
```

**Comment by ada4a** [comment]
Fwiw the introduced-inconsistency thing is a problem that some other lints have as well, but it's not clear how a lint should recognize that.

I do like the proposal, but I'm afraid it would also stop things like
```diff
- Duration::from_secs(5 * 60 * 60)
+ Duration::from_hours(5)
```
which is imo unfortunate.

I'd say the examples you list sound like good candidates for sprinkling some `#[allow]`s

**Comment by ensc** [comment]
> I'm afraid it would also stop things like
> 
> ```diff
> - Duration::from_secs(5 * 60 * 60)
> + Duration::from_hours(5)
> ```
>
> which is imo unfortunate.

Does clippy not have information whether the input is a literal (--> no change) or an expression?

**Comment by ada4a** [comment]
It does! So, are you proposing to:
- on literals, only apply the transformation if the resulting value will be >10
- on expressions, apply the transformation always

? I think that's a good idea, as it reduces FPs while not creating too many FNs.

This enhancement would be reasonably easy to implement I think, so I'd mark this as a good first issue. You're welcome to pick it up if you want:)

@rustbot label "good first issue"

**Comment by hriztam** [comment]
Hello, if this issue is not fixed yet, I would like to work on it @ensc

**Comment by ada4a** [comment]
Thank you for your interest @hriztam – but it looks like there are already a PR open for this issue (two, even), so I'd say it's already taken care of

**Comment by arunsingh** [comment]
Hi @ensc , I am interested in helping with this. I see this is currently assigned, so I do not want to duplicate work. If the assignee is no longer working on it, I can take a small, targeted pass with UI tests and a minimal diff.
